### DISCUSSION TITLE
`yarn workspaces foreach` doesn't work with Github Actions, how should the problem be raised?

### DISCUSSION THREAD

**Discussion by SmileyJames** [yarnpkg/berry#2650] [Q&A] [answered]
This foreach command normally executes a number of tests in the mono repo. When running on Github Actons, only one test runs:
```
5s
Run yarn workspaces foreach run test
➤ YN0000: ➤ YN0000: (node:1859) [MODULE_NOT_FOUND] Error: debug tried to access supports-color (a peer dependency) but it isn't provided by its ancestors; this makes the require call ambiguous and unsound.
➤ YN0000: ➤ YN0000: (Use `node --trace-warnings ...` to show where the warning was created)
➤ YN0000: ➤ YN0000: PASS src/index.test.js
➤ YN0000: ➤ YN0000:   ✓ pass (1 ms)
➤ YN0000: ➤ YN0000: 
➤ YN0000: ➤ YN0000: Test Suites: 1 passed, 1 total
➤ YN0000: ➤ YN0000: Tests:       1 passed, 1 total
➤ YN0000: ➤ YN0000: Snapshots:   0 total
➤ YN0000: ➤ YN0000: Time:        1.981 s
➤ YN0000: ➤ YN0000: Ran all test suites.
```

Searching Google and Github issues for people with similiar issues has not yielded results.

However, the yarn mono repo uses Github actions and does not use `yarn workspaces foreach`, instead opting for [a long list of packages hardcoded to the build script](https://github.com/yarnpkg/berry/blob/master/.github/workflows/integration-workflow.yml#L118)

Everything works well for me locally (on MacOS, the GH Actions are running on Ubuntu) so I am unsure how to raise the issue. It's not clear to me if this is a problem with [yarn workspace-tools source code](https://github.com/yarnpkg/berry/blob/master/packages/plugin-workspace-tools/sources/commands/foreach.ts) or more of an integration issue with GH actions.

**Comment by merceyz** [comment, marked as answer]
If I were to guess your CI is using Node 15 and you're not. There is a bug with Node 15 and Yarn that is [fixed in `master`](https://github.com/yarnpkg/berry/pull/2255) so you can try `yarn set version from sources`, make sure to import any plugins you use from sources as well.

https://yarnpkg.com/cli/set/version/from/sources
https://yarnpkg.com/cli/plugin/import/from/sources

**Reply by SmileyJames** [reply]
Ahh, that would explain it. Using 14 locally and 15 on CI. Thank you 😃
